demo/admin/src/products/future/ProductForm.cometGen.ts

```diff
@@ -26,6 +26,16 @@ export const ProductForm: FormConfig<GQLProduct> = {
                 { type: "text", name: "description", label: "Description", multiline: true },
                 { type: "staticSelect", name: "type", label: "Type", required: true /*, values: from gql schema (TODO overridable)*/ },
                 { type: "asyncSelect", name: "category", rootQuery: "productCategories" },
+                {
+                    type: "optionalFields",
+                    name: "dimensions",
+                    checkboxLabel: "Configure dimensions",
+                    fields: [
+                        { type: "number", name: "width", label: "Width" },
```

Comment:
the field is actually dimensions.width, but you are omitting the dimensions for optionalFields

- this might be unexpected (DX)?
- you might want to have non-nested fields?
- if we agree on the current api, the type might be changed to "optionalNestedFields" or so

Reply:
you're right. for now only nested fields are required to be wrapped in optionalFields. And as we discussed earlier (in person) optionalFields might also be one of the only cases where nesting is really required (except for probably readonly/show/display fields, or something like cards in koala)

I'll rename it to `optionalNestedFields` because optionalFields with non-nested fields need a different code/implementation (unsetting by field instead of unsetting the object, probably always saving `xxxEnabled`). And `optionalNestedFields` is what I've implemented.